From 436599c3d935501592ed57148751d35474439d25 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 10 Aug 2018 16:57:06 +0300
Subject: [PATCH] hl: preperation to Gaudi step 2

replace hard coded "dali" with "$(HL_DEV)" as much as possible
also move to common scripts that can be reused
---

diff --git a/Makefile b/Makefile
index 19ed016..967034b 100644
--- a/Makefile
+++ b/Makefile
@@ -97,20 +97,20 @@
 dali-asic-install: DALI_INSTALL_HOST_IP ?= 192.168.10.134
 .PHONY: dali-install
 dali-install:
-	-sshpass -p123456        scp $(IMAGES_DIR)/dali-u-boot.bin	root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/dali/
-	-sshpass -p123456        scp $(IMAGES_DIR)/dali-fit.itb		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/dali/
-	-sshpass -p123456        scp $(IMAGES_DIR)/dali-spi-flash.bin	root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/dali/
-	-sshpass -p123456        scp $(IMAGES_DIR)/fw_loader		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/dali/
-	-sshpass -p123456        scp $(IMAGES_DIR)/dali-u-boot.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/dali/
-	-sshpass -p123456        scp $(IMAGES_DIR)/dali-u-boot-spl.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/dali/
-	-sshpass -p123456        scp $(IMAGES_DIR)/dali-fit.itb		root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/
-	-sshpass -p123456        scp $(IMAGES_DIR)/boot.scr		root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-	-sshpass -p123456        scp $(IMAGES_DIR)/fw_loader		root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-	-sshpass -p123456        scp $(IMAGES_DIR)/dali-spi-flash.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/dali/
-	-sshpass -p123456        scp $(UBOOT_BASE)/u-boot		root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/dali/
-	-sshpass -p123456        scp $(LINUX_BASE)/vmlinux		root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/dali/
-	-sshpass -p123456        scp -r $(UBOOT_XMPL)/dali_*.bin        root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-	-sshpass -p123456        scp ${DSV_BASE}/dsv.bin		root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/fw_loader			root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot-spl.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/
+	-sshpass -p123456        scp $(IMAGES_DIR)/boot.scr			root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	-sshpass -p123456        scp $(IMAGES_DIR)/fw_loader			root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(UBOOT_BASE)/u-boot			root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(LINUX_BASE)/vmlinux			root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp -r $(UBOOT_XMPL)/$(HL_DEV)*.bin        	root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	-sshpass -p123456        scp ${DSV_BASE}/dsv.bin			root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
 dali-onix-install dali-pldm-install dali-asic-install: dali-install
 dali-onix-install:
 	@
@@ -118,15 +118,15 @@
 ${PLDM_HEX_DIR}:
 	@mkdir -p $@
 dali-pldm-install: ${PLDM_HEX_DIR}
-	-cp $(IMAGES_DIR)/dali-spi-flash.hex $</
+	-cp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.hex $</
 ASIC_BIN_DIR ?= /tools/hl_shared_dir/production/embedded/head
 ${ASIC_BIN_DIR}:
 	@mkdir -p $@
 dali-asic-install: ${ASIC_BIN_DIR}
-	-cp $(IMAGES_DIR)/dali-fit.itb $</
-	-cp $(IMAGES_DIR)/dali-spi-flash.bin $</
-	-cp $(IMAGES_DIR)/dali-u-boot.bin $</
-	-cp $(IMAGES_DIR)/dali-u-boot-spl.bin $</
+	-cp $(IMAGES_DIR)/$(HL_DEV)-fit.itb $</
+	-cp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin $</
+	-cp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin $</
+	-cp $(IMAGES_DIR)/$(HL_DEV)-u-boot-spl.bin $</
 	-cp $(IMAGES_DIR)/fw_loader $</
 ifneq ($(notdir $(realpath $(ASIC_BIN_DIR))), head)
 	ln -nfs $$(basename $(ASIC_BIN_DIR)) $$(dirname $(ASIC_BIN_DIR))/latest
@@ -142,7 +142,7 @@
 .PHONY: dali-onix-test dali-pldm-test dali-asic-test
 dali-onix-test: dali-onix-tools-install
 dali-onix-test dali-pldm-test dali-asic-test:
-	@HOST_UBOOT_TEST_HOOKS_BRD=$(patsubst dali-%-test,dali_%,$@)	\
+	@HOST_UBOOT_TEST_HOOKS_BRD=$(patsubst $(HL_DEV)-%-test,$(HL_DEV)_%,$@)	\
 		$(MAKE) host-uboot-test-hooks-rebuild
 
 dali-onix-tools-install: DALI_INSTALL_LAB_IP ?= monster4-nuc
@@ -164,9 +164,9 @@
 dali-asic-tbot: HOST_TBOT_IP ?= 192.168.10.109
 dali-asic-tbot: PCIE_HOST_TBOT_IP ?= monster5
 dali-onix-tbot dali-pldm-tbot dali-asic-tbot:
-	@HOST_TBOT_LAB=lab_dali					\
-	HOST_TBOT_CFG=dali					\
-	HOST_TBOT_BRD=$(patsubst dali-%-tbot,%,$@)		\
+	@HOST_TBOT_LAB=lab_$(HL_DEV)					\
+	HOST_TBOT_CFG=$(HL_DEV)					\
+	HOST_TBOT_BRD=$(patsubst $(HL_DEV)-%-tbot,%,$@)		\
 	HOST_TBOT_CASE=$(HOST_TBOT_TC)				\
 	HOST_TBOT_LAB_IP=$(HOST_TBOT_IP)			\
 	PCIE_HOST_TBOT_LAB_IP=$(PCIE_HOST_TBOT_IP)		\
diff --git a/board/habanalabs/common/busybox.fragment b/board/habanalabs/common/busybox.fragment
new file mode 100644
index 0000000..8edd83c
--- /dev/null
+++ b/board/habanalabs/common/busybox.fragment
@@ -0,0 +1,25 @@
+CONFIG_FIND=y
+CONFIG_FEATURE_FIND_PRINT0=y
+CONFIG_MKTEMP=y
+CONFIG_MODPROBE=y
+CONFIG_DEFAULT_MODULES_DIR="/lib/modules"
+CONFIG_DEFAULT_DEPMOD_FILE="modules.dep"
+CONFIG_DIFF=y
+CONFIG_CKSUM=y
+CONFIG_WATCH=y
+CONFIG_VI=y
+CONFIG_FEATURE_VI_MAX_LEN=4096
+CONFIG_FEATURE_VI_COLON=y
+CONFIG_FEATURE_VI_YANKMARK=y
+CONFIG_FEATURE_VI_SEARCH=y
+CONFIG_FEATURE_VI_USE_SIGNALS=y
+CONFIG_FEATURE_VI_DOT_CMD=y
+CONFIG_FEATURE_VI_READONLY=y
+CONFIG_FEATURE_VI_SETOPTS=y
+CONFIG_FEATURE_VI_SET=y
+CONFIG_FEATURE_VI_WIN_RESIZE=y
+CONFIG_FEATURE_VI_ASK_TERMINAL=y
+CONFIG_FEATURE_VI_UNDO=y
+CONFIG_FEATURE_VI_UNDO_QUEUE=y
+CONFIG_FEATURE_VI_UNDO_QUEUE_MAX=256
+CONFIG_FEATURE_ALLOW_EXEC=y
diff --git a/board/habanalabs/common/hl_fit_spl.its b/board/habanalabs/common/hl_fit_spl.its
new file mode 100644
index 0000000..9ce365d
--- /dev/null
+++ b/board/habanalabs/common/hl_fit_spl.its
@@ -0,0 +1,72 @@
+/*
+ * (C) Copyright 2017 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+/dts-v1/;
+
+/ {
+	description = "FIT image with U-Boot proper, DTB";
+	#address-cells = <2>;
+
+	images {
+		uboot {
+			description = "U-Boot (64-bit)";
+			data = /incbin/("u-boot-nodtb.bin");
+			type = "standalone";
+			arch = "arm64";
+			compression = "none";
+			load = <0x7f 0xf0100000>;
+		};
+
+		fdt {
+			description = "Dali 1000 flat device-tree";
+			data = /incbin/("u-boot.dtb");
+			type = "flat_dt";
+			compression = "none";
+		};
+/*
+		data0 {
+			description = "Dali data 0";
+			data = /incbin/("uboot-data0.bin");
+			compression = "none";
+			load = <0x7f 0xf0000000>;
+		};
+
+		data1 {
+			description = "Dali data 1";
+			data = /incbin/("uboot-data1.bin");
+			compression = "none";
+			load = <0x7f 0xf0010000>;
+		};
+
+		data2 {
+			description = "Dali data 2";
+			data = /incbin/("uboot-data2.bin");
+			compression = "none";
+			load = <0x7f 0xf0020000>;
+		};
+
+		data3 {
+			description = "Dali data 3";
+			data = /incbin/("uboot-data3.bin");
+			compression = "none";
+			load = <0x7f 0xf0030000>;
+		};
+*/
+	};
+
+	configurations {
+		default = "conf";
+		conf {
+			description = "Habanalabs Dali";
+			firmware = "uboot";
+			fdt = "fdt";
+/*
+			loadables = "data0", "data1", "data2", "data3";
+*/
+		};
+	};
+};
diff --git a/board/habanalabs/common/post-build.sh b/board/habanalabs/common/post-build.sh
new file mode 100755
index 0000000..493187c
--- /dev/null
+++ b/board/habanalabs/common/post-build.sh
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+pushd ${TARGET_DIR}/etc/init.d/
+	if [ "${HL_TARGET}" == "pldm" ] || [ "${HL_TARGET}" == "asic" ]; then
+		if [ -e S40network ]; then
+			mv S40network .S40network
+		fi
+	elif [ "${HL_TARGET}" == "onix" ]; then
+		if [ -e .S40network ]; then
+			mv .S40network S40network
+		fi
+	else
+		echo "Unknown ${HL_DEV} target: ${HL_TARGET}"
+		exit 1
+	fi
+popd
+
+# Add armcpd to hl device
+if [ -e ${TARGET_DIR}/etc/inittab ]; then
+    grep -qE "^${HL_DEV}::" ${TARGET_DIR}/etc/inittab || \
+	sed -i "/GENERIC_SERIAL/a# Put armcpd on Dali device/\n\
+${HL_DEV}::respawn:/usr/bin/armcpd # ARM CP" ${TARGET_DIR}/etc/inittab
+fi
diff --git a/board/habanalabs/common/post-image.sh b/board/habanalabs/common/post-image.sh
new file mode 100755
index 0000000..b4caf7a
--- /dev/null
+++ b/board/habanalabs/common/post-image.sh
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+BOARD_BASE=${BR2_EXTERNAL_HL_DALI_PATH}/board/habanalabs/${HL_DEV}
+SPL_BINFILE=${SPL_BINFILE:-u-boot-spl.bin}
+UBOOT_CMDFILE=${UBOOT_CMDFILE:-${BOARD_BASE}/uboot-eeprom.cmd}
+UBOOT_FITFILE=u-boot.itb
+UBOOT_ITSFILE=${UBOOT_ITSFILE:-${BOARD_BASE}/../common/hl_fit_spl.its}
+UBOOT_BINFILE=${UBOOT_BINFILE:-${UBOOT_FITFILE}}
+LINUX_FITFILE=fit.itb
+
+function swap_file {
+	INF=$1
+	OUTF=$2
+
+	# SWAP needs this to be 4B aligned in size
+	dd status=none conv=sync bs=4 if=${INF} > ${OUTF}.tmp
+	# Sadly --endian=big is not swapping for -tx1 so I used some gawk magic
+	od -Anone -v -tx1 -w4 ${OUTF}.tmp | gawk '{ print $4 $3 $2 $1 }' | xxd -r -p > ${OUTF}
+}
+
+pushd ${BINARIES_DIR}
+	cp ${BOARD_BASE}/${HL_DEV}_${HL_TARGET}.its .
+	cp ${BASE_DIR}/build/linux-custom/arch/arm64/boot/Image .
+	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${HL_DEV}_${HL_TARGET}.its ${LINUX_FITFILE}
+
+	cp ${UBOOT_ITSFILE} ${HL_DEV}_fit_spl.its
+	# In case ITSFILE points to locations under public directory /tools
+	ln -nfs /tools
+	# SPI IOMAP likes these to be 8B aligned in size
+	dd status=none conv=sync bs=8 if=${BASE_DIR}/build/uboot-custom/u-boot-nodtb.bin	 > u-boot-nodtb.bin
+	dd status=none conv=sync bs=8 if=${BASE_DIR}/build/uboot-custom/u-boot.dtb		 > u-boot.dtb
+	${HOST_DIR}/usr/bin/mkimage -f ${HL_DEV}_fit_spl.its -E ${UBOOT_FITFILE}
+	${HOST_DIR}/usr/bin/mkimage -C none -A arm64 -T script -d ${UBOOT_CMDFILE} boot.scr
+
+	dd status=none conv=sync bs=8 if=${SPL_BINFILE}						 > ${HL_DEV}-u-boot-spl.bin
+	# Swap SPL
+	swap_file ${HL_DEV}-u-boot-spl.bin ${HL_DEV}-u-boot-spl.bin.be
+	SPL_BINFILE=${HL_DEV}-u-boot-spl.bin.be
+
+	# KMD like this to be 4B aligned in size
+	dd status=none conv=sync bs=4 if=${LINUX_FITFILE}					 > ${HL_DEV}-fit.itb
+	# Used by TRACE32 (Lauterbach JTAG)
+	dd status=none conv=sync bs=4 if=u-boot.bin						 > ${HL_DEV}-u-boot.bin
+
+	### Construct bin file
+	# BTL likes these to be 16B aligned in size
+	dd status=none conv=sync bs=16 if=preboot.bin						 > ${HL_DEV}-preboot.bin
+	# prepare BTL image
+	BTL_CMD_ARGS="${BTL_CMD_ARGS} SPL_SIZE=$(printf "0x%X\n" `stat -c "%s" ${SPL_BINFILE}`)"
+	BTL_CMD_ARGS="${BTL_CMD_ARGS} UBOOT_SIZE=$(printf "0x%X\n" `stat -c "%s" ${UBOOT_BINFILE}`)"
+	cp -a ${BOARD_BASE}/btl .
+	make -C btl clean all ${BTL_CMD_ARGS}
+	cp btl/sectors-be.bin .
+
+	# Metadata
+	dd status=none conv=sync bs=64K count=1 if=/dev/zero				 > ${HL_DEV}-pkey-hash.bin
+	dd status=none conv=sync bs=64K count=1 if=/dev/zero				 > ${HL_DEV}-ver.bin
+	echo "11110000" > ${HL_DEV}-next-image.txt
+	dd status=none conv=sync bs=64K if=${HL_DEV}-next-image.txt			 > ${HL_DEV}-next-image.bin
+	dd status=none conv=sync bs=64K count=1 if=/dev/zero				 > ${HL_DEV}-images-pointers.bin
+
+	# BTL
+	dd status=none conv=sync bs=256K	if=sectors-be.bin			 > ${HL_DEV}-spi-flash.bin.nocrc
+	# Metadata
+	dd status=none conv=sync bs=64K		if=${HL_DEV}-pkey-hash.bin		>> ${HL_DEV}-spi-flash.bin.nocrc
+	dd status=none conv=sync bs=64K		if=${HL_DEV}-ver.bin			>> ${HL_DEV}-spi-flash.bin.nocrc
+	dd status=none conv=sync bs=64K		if=${HL_DEV}-next-image.bin		>> ${HL_DEV}-spi-flash.bin.nocrc
+	dd status=none conv=sync bs=64K		if=${HL_DEV}-images-pointers.bin	>> ${HL_DEV}-spi-flash.bin.nocrc
+	# U-Boot ENV
+	dd status=none conv=sync bs=64K		if=uboot-env.bin			>> ${HL_DEV}-spi-flash.bin.nocrc
+	# U-Boot ENV Redundand
+	dd status=none conv=sync bs=64K		if=uboot-env.bin			>> ${HL_DEV}-spi-flash.bin.nocrc
+	# SPL
+	dd status=none conv=sync bs=64K		if=${SPL_BINFILE}			>> ${HL_DEV}-spi-flash.bin.nocrc
+	# U-Boot
+	dd status=none conv=sync bs=1M		if=${UBOOT_BINFILE}			>> ${HL_DEV}-spi-flash.bin.nocrc
+
+	# Prepare checksum
+	CKSUM=`cksum ${HL_DEV}-spi-flash.bin.nocrc | cut -d' ' -f1`
+	CKSUM_BE=`printf "%08x" ${CKSUM}`
+	CKSUM_LE=${CKSUM_BE:6:2}${CKSUM_BE:4:2}${CKSUM_BE:2:2}${CKSUM_BE:0:2}
+	echo ${CKSUM_LE} | xxd -p -r							 > ${HL_DEV}-spi-flash.crc
+	dd status=none conv=sync bs=64K		if=${HL_DEV}-spi-flash.bin.nocrc	 > ${HL_DEV}-spi-flash.bin
+	dd status=none conv=sync bs=4		if=${HL_DEV}-spi-flash.crc		>> ${HL_DEV}-spi-flash.bin
+
+	# Convert bin file into Palladium hex format
+	xxd -p -c1 -g1				${HL_DEV}-spi-flash.bin	 		 > ${HL_DEV}-spi-flash.hex
+popd
diff --git a/board/habanalabs/dali/busybox.fragment b/board/habanalabs/dali/busybox.fragment
deleted file mode 100644
index 8edd83c..0000000
--- a/board/habanalabs/dali/busybox.fragment
+++ /dev/null
@@ -1,25 +0,0 @@
-CONFIG_FIND=y
-CONFIG_FEATURE_FIND_PRINT0=y
-CONFIG_MKTEMP=y
-CONFIG_MODPROBE=y
-CONFIG_DEFAULT_MODULES_DIR="/lib/modules"
-CONFIG_DEFAULT_DEPMOD_FILE="modules.dep"
-CONFIG_DIFF=y
-CONFIG_CKSUM=y
-CONFIG_WATCH=y
-CONFIG_VI=y
-CONFIG_FEATURE_VI_MAX_LEN=4096
-CONFIG_FEATURE_VI_COLON=y
-CONFIG_FEATURE_VI_YANKMARK=y
-CONFIG_FEATURE_VI_SEARCH=y
-CONFIG_FEATURE_VI_USE_SIGNALS=y
-CONFIG_FEATURE_VI_DOT_CMD=y
-CONFIG_FEATURE_VI_READONLY=y
-CONFIG_FEATURE_VI_SETOPTS=y
-CONFIG_FEATURE_VI_SET=y
-CONFIG_FEATURE_VI_WIN_RESIZE=y
-CONFIG_FEATURE_VI_ASK_TERMINAL=y
-CONFIG_FEATURE_VI_UNDO=y
-CONFIG_FEATURE_VI_UNDO_QUEUE=y
-CONFIG_FEATURE_VI_UNDO_QUEUE_MAX=256
-CONFIG_FEATURE_ALLOW_EXEC=y
diff --git a/board/habanalabs/dali/dali_fit_spl.its b/board/habanalabs/dali/dali_fit_spl.its
deleted file mode 100644
index 9ce365d..0000000
--- a/board/habanalabs/dali/dali_fit_spl.its
+++ /dev/null
@@ -1,72 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-/dts-v1/;
-
-/ {
-	description = "FIT image with U-Boot proper, DTB";
-	#address-cells = <2>;
-
-	images {
-		uboot {
-			description = "U-Boot (64-bit)";
-			data = /incbin/("u-boot-nodtb.bin");
-			type = "standalone";
-			arch = "arm64";
-			compression = "none";
-			load = <0x7f 0xf0100000>;
-		};
-
-		fdt {
-			description = "Dali 1000 flat device-tree";
-			data = /incbin/("u-boot.dtb");
-			type = "flat_dt";
-			compression = "none";
-		};
-/*
-		data0 {
-			description = "Dali data 0";
-			data = /incbin/("uboot-data0.bin");
-			compression = "none";
-			load = <0x7f 0xf0000000>;
-		};
-
-		data1 {
-			description = "Dali data 1";
-			data = /incbin/("uboot-data1.bin");
-			compression = "none";
-			load = <0x7f 0xf0010000>;
-		};
-
-		data2 {
-			description = "Dali data 2";
-			data = /incbin/("uboot-data2.bin");
-			compression = "none";
-			load = <0x7f 0xf0020000>;
-		};
-
-		data3 {
-			description = "Dali data 3";
-			data = /incbin/("uboot-data3.bin");
-			compression = "none";
-			load = <0x7f 0xf0030000>;
-		};
-*/
-	};
-
-	configurations {
-		default = "conf";
-		conf {
-			description = "Habanalabs Dali";
-			firmware = "uboot";
-			fdt = "fdt";
-/*
-			loadables = "data0", "data1", "data2", "data3";
-*/
-		};
-	};
-};
diff --git a/board/habanalabs/dali/post-build.sh b/board/habanalabs/dali/post-build.sh
deleted file mode 100755
index c7cb9e1..0000000
--- a/board/habanalabs/dali/post-build.sh
+++ /dev/null
@@ -1,23 +0,0 @@
-#!/usr/bin/env bash
-
-pushd ${TARGET_DIR}/etc/init.d/
-	if [ "${HL_TARGET}" == "pldm" ] || [ "${HL_TARGET}" == "asic" ]; then
-		if [ -e S40network ]; then
-			mv S40network .S40network
-		fi
-	elif [ "${HL_TARGET}" == "onix" ]; then
-		if [ -e .S40network ]; then
-			mv .S40network S40network
-		fi
-	else
-		echo "Unknown ${HL_DEV} target: ${HL_TARGET}"
-		exit 1
-	fi
-popd
-
-# Add armcpd to dali
-if [ -e ${TARGET_DIR}/etc/inittab ]; then
-    grep -qE '^dali::' ${TARGET_DIR}/etc/inittab || \
-	sed -i '/GENERIC_SERIAL/a# Put armcpd on Dali device\
-dali::respawn:/usr/bin/armcpd # ARM CP' ${TARGET_DIR}/etc/inittab
-fi
diff --git a/board/habanalabs/dali/post-image.sh b/board/habanalabs/dali/post-image.sh
deleted file mode 100755
index 34a94af..0000000
--- a/board/habanalabs/dali/post-image.sh
+++ /dev/null
@@ -1,88 +0,0 @@
-#!/usr/bin/env bash
-
-BOARD_BASE=${BR2_EXTERNAL_HL_DALI_PATH}/board/habanalabs/dali
-SPL_BINFILE=${SPL_BINFILE:-u-boot-spl.bin}
-UBOOT_CMDFILE=${UBOOT_CMDFILE:-${BOARD_BASE}/uboot-eeprom.cmd}
-UBOOT_FITFILE=u-boot.itb
-UBOOT_ITSFILE=${UBOOT_ITSFILE:-${BOARD_BASE}/dali_fit_spl.its}
-UBOOT_BINFILE=${UBOOT_BINFILE:-${UBOOT_FITFILE}}
-LINUX_FITFILE=fit.itb
-
-function swap_file {
-	INF=$1
-	OUTF=$2
-
-	# SWAP needs this to be 4B aligned in size
-	dd status=none conv=sync bs=4 if=${INF} > ${OUTF}.tmp
-	# Sadly --endian=big is not swapping for -tx1 so I used some gawk magic
-	od -Anone -v -tx1 -w4 ${OUTF}.tmp | gawk '{ print $4 $3 $2 $1 }' | xxd -r -p > ${OUTF}
-}
-
-pushd ${BINARIES_DIR}
-	cp ${BOARD_BASE}/dali_${HL_TARGET}.its .
-	cp ${BASE_DIR}/build/linux-custom/arch/arm64/boot/Image .
-	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f dali_${HL_TARGET}.its ${LINUX_FITFILE}
-
-	cp ${UBOOT_ITSFILE} dali_fit_spl.its
-	# In case ITSFILE points to locations under public directory /tools
-	ln -nfs /tools
-	# SPI IOMAP likes these to be 8B aligned in size
-	dd status=none conv=sync bs=8 if=${BASE_DIR}/build/uboot-custom/u-boot-nodtb.bin	 > u-boot-nodtb.bin
-	dd status=none conv=sync bs=8 if=${BASE_DIR}/build/uboot-custom/u-boot.dtb		 > u-boot.dtb
-	${HOST_DIR}/usr/bin/mkimage -f dali_fit_spl.its -E ${UBOOT_FITFILE}
-	${HOST_DIR}/usr/bin/mkimage -C none -A arm64 -T script -d ${UBOOT_CMDFILE} boot.scr
-
-	dd status=none conv=sync bs=8 if=${SPL_BINFILE}						 > dali-u-boot-spl.bin
-	# Swap SPL
-	swap_file dali-u-boot-spl.bin dali-u-boot-spl.bin.be
-	SPL_BINFILE=dali-u-boot-spl.bin.be
-
-	# KMD like this to be 4B aligned in size
-	dd status=none conv=sync bs=4 if=${LINUX_FITFILE}					 > dali-fit.itb
-	# Used by TRACE32 (Lauterbach JTAG)
-	dd status=none conv=sync bs=4 if=u-boot.bin						 > dali-u-boot.bin
-
-	### Construct bin file
-	# BTL likes these to be 16B aligned in size
-	dd status=none conv=sync bs=16 if=preboot.bin						 > dali-preboot.bin
-	# prepare BTL image
-	BTL_CMD_ARGS="${BTL_CMD_ARGS} SPL_SIZE=$(printf "0x%X\n" `stat -c "%s" ${SPL_BINFILE}`)"
-	BTL_CMD_ARGS="${BTL_CMD_ARGS} UBOOT_SIZE=$(printf "0x%X\n" `stat -c "%s" ${UBOOT_BINFILE}`)"
-	cp -a ${BOARD_BASE}/btl .
-	make -C btl clean all ${BTL_CMD_ARGS}
-	cp btl/sectors-be.bin .
-
-	# Metadata
-	dd status=none conv=sync bs=64K count=1 if=/dev/zero					 > dali-pkey-hash.bin
-	dd status=none conv=sync bs=64K count=1 if=/dev/zero					 > dali-ver.bin
-	echo "11110000" > dali-next-image.txt
-	dd status=none conv=sync bs=64K if=dali-next-image.txt					 > dali-next-image.bin
-	dd status=none conv=sync bs=64K count=1 if=/dev/zero					 > dali-images-pointers.bin
-
-	# BTL
-	dd status=none conv=sync bs=256K	if=sectors-be.bin				 > dali-spi-flash.bin.nocrc
-	# Metadata
-	dd status=none conv=sync bs=64K		if=dali-pkey-hash.bin				>> dali-spi-flash.bin.nocrc
-	dd status=none conv=sync bs=64K		if=dali-ver.bin					>> dali-spi-flash.bin.nocrc
-	dd status=none conv=sync bs=64K		if=dali-next-image.bin				>> dali-spi-flash.bin.nocrc
-	dd status=none conv=sync bs=64K		if=dali-images-pointers.bin			>> dali-spi-flash.bin.nocrc
-	# U-Boot ENV
-	dd status=none conv=sync bs=64K		if=uboot-env.bin				>> dali-spi-flash.bin.nocrc
-	# U-Boot ENV Redundand
-	dd status=none conv=sync bs=64K		if=uboot-env.bin				>> dali-spi-flash.bin.nocrc
-	# SPL
-	dd status=none conv=sync bs=64K		if=${SPL_BINFILE}				>> dali-spi-flash.bin.nocrc
-	# U-Boot
-	dd status=none conv=sync bs=1M		if=${UBOOT_BINFILE}				>> dali-spi-flash.bin.nocrc
-
-	# Prepare checksum
-	CKSUM=`cksum dali-spi-flash.bin.nocrc | cut -d' ' -f1`
-	CKSUM_BE=`printf "%08x" ${CKSUM}`
-	CKSUM_LE=${CKSUM_BE:6:2}${CKSUM_BE:4:2}${CKSUM_BE:2:2}${CKSUM_BE:0:2}
-	echo ${CKSUM_LE} | xxd -p -r								 > dali-spi-flash.crc
-	dd status=none conv=sync bs=64K		if=dali-spi-flash.bin.nocrc			 > dali-spi-flash.bin
-	dd status=none conv=sync bs=4		if=dali-spi-flash.crc				>> dali-spi-flash.bin
-
-	# Convert bin file into Palladium hex format
-	xxd -p -c1 -g1				dali-spi-flash.bin	 			 > dali-spi-flash.hex
-popd
diff --git a/configs/dali_defconfig b/configs/dali_defconfig
index 66049d2..7c41127 100644
--- a/configs/dali_defconfig
+++ b/configs/dali_defconfig
@@ -3,24 +3,24 @@
 BR2_DL_DIR="$(BR2_EXTERNAL_HL_DALI_PATH)/dl"
 BR2_PACKAGE_OVERRIDE_FILE="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/common/local.mk"
 BR2_TOOLCHAIN_EXTERNAL=y
-BR2_TARGET_GENERIC_HOSTNAME="dali"
+BR2_TARGET_GENERIC_HOSTNAME="$(HL_DEV)"
 BR2_TARGET_GENERIC_ISSUE="Welcome to Dali"
 BR2_ROOTFS_DEVICE_CREATION_DYNAMIC_MDEV=y
 BR2_TARGET_GENERIC_ROOT_PASSWD="root"
 BR2_SYSTEM_BIN_SH_BASH=y
 BR2_SYSTEM_DHCP="eth0"
 BR2_ROOTFS_OVERLAY="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/common/rootfs_overlay"
-BR2_ROOTFS_POST_BUILD_SCRIPT="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/dali/post-build.sh"
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/dali/post-image.sh"
+BR2_ROOTFS_POST_BUILD_SCRIPT="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/common/post-build.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/common/post-image.sh"
 BR2_LINUX_KERNEL=y
 BR2_LINUX_KERNEL_USE_ARCH_DEFAULT_CONFIG=y
-BR2_LINUX_KERNEL_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/dali/linux_$(HL_TARGET).fragment"
+BR2_LINUX_KERNEL_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/$(HL_DEV)/linux_$(HL_TARGET).fragment"
 BR2_LINUX_KERNEL_IMAGE_TARGET_CUSTOM=y
 BR2_LINUX_KERNEL_IMAGE_TARGET_NAME="Image.gz"
 BR2_LINUX_KERNEL_DTS_SUPPORT=y
-BR2_LINUX_KERNEL_INTREE_DTS_NAME="habanalabs/dali_$(HL_TARGET)"
+BR2_LINUX_KERNEL_INTREE_DTS_NAME="habanalabs/$(HL_DEV)_$(HL_TARGET)"
 BR2_PACKAGE_BUSYBOX_CONFIG="package/busybox/busybox-minimal.config"
-BR2_PACKAGE_BUSYBOX_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/dali/busybox.fragment"
+BR2_PACKAGE_BUSYBOX_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/common/busybox.fragment"
 BR2_PACKAGE_BUSYBOX_SHOW_OTHERS=y
 BR2_PACKAGE_BUSYBOX_WATCHDOG=y
 BR2_PACKAGE_MTD=y
@@ -48,12 +48,12 @@
 # BR2_TARGET_ROOTFS_TAR is not set
 BR2_TARGET_UBOOT=y
 BR2_TARGET_UBOOT_BUILD_SYSTEM_KCONFIG=y
-BR2_TARGET_UBOOT_BOARD_DEFCONFIG="dali"
-BR2_TARGET_UBOOT_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/dali/uboot_$(HL_TARGET).fragment"
+BR2_TARGET_UBOOT_BOARD_DEFCONFIG="$(HL_DEV)"
+BR2_TARGET_UBOOT_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/$(HL_DEV)/uboot_$(HL_TARGET).fragment"
 BR2_TARGET_UBOOT_NEEDS_DTC=y
 BR2_TARGET_UBOOT_SPL=y
 BR2_TARGET_UBOOT_ENVIMAGE=y
-BR2_TARGET_UBOOT_ENVIMAGE_SOURCE="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/dali/uboot-env-$(HL_TARGET).txt"
+BR2_TARGET_UBOOT_ENVIMAGE_SOURCE="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/$(HL_DEV)/uboot-env-$(HL_TARGET).txt"
 BR2_TARGET_UBOOT_ENVIMAGE_SIZE="0x1000"
 BR2_TARGET_UBOOT_ENVIMAGE_REDUNDANT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS=y
diff --git a/package/uboot-test-hooks/uboot-test-hooks.mk b/package/uboot-test-hooks/uboot-test-hooks.mk
index fe403e2..bee748a 100644
--- a/package/uboot-test-hooks/uboot-test-hooks.mk
+++ b/package/uboot-test-hooks/uboot-test-hooks.mk
@@ -7,7 +7,7 @@
 HOST_UBOOT_TEST_HOOKS_MAKE_OPTS += --build-dir=$(UBOOT_DIR)
 HOST_UBOOT_TEST_HOOKS_MAKE_OPTS += --board-type=$(HOST_UBOOT_TEST_HOOKS_BRD)
 HOST_UBOOT_TEST_HOOKS_MAKE_OPTS +=  -s
-HOST_UBOOT_TEST_HOOKS_MAKE_OPTS +=  -k dali
+HOST_UBOOT_TEST_HOOKS_MAKE_OPTS +=  -k $(HL_DEV)
 HOST_UBOOT_TEST_HOOKS_MAKE_ENV   += PATH=$(PATH):$(@D)/bin
 HOST_UBOOT_TEST_HOOKS_MAKE_ENV   += PYTHONPATH=$(PYTHONPATH):$(@D)/py/habanalabs
 
